app/src/main/java/ar/com/wolox/android/example/network/repository/NewsRepository.kt

```diff
@@ -0,0 +1,17 @@
+package ar.com.wolox.android.example.network.repository
+
+import ar.com.wolox.android.example.network.services.NewsService
+import ar.com.wolox.wolmo.networking.retrofit.RetrofitServices
+import ar.com.wolox.wolmo.networking.retrofit.handler.NetworkRequestHandler
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class NewsRepository @Inject constructor (private val retrofitServices: RetrofitServices) {
+    private val services: NewsService
+        get() = retrofitServices.getService(NewsService::class.java)
+
+    suspend fun getNewsList(page: Int?) = withContext(Dispatchers.IO) {
+        page?.let { NetworkRequestHandler.safeApiCall { services.getNewsPage(page) } } ?: NetworkRequestHandler.safeApiCall { services.getFirstPage() }
```

Comment:
You can set a default value for page instead of using this elvis operator. Something like `suspend fun getNewsList(page: Int? = 1)`